app/routes.py

```diff
@@ -1,2 +1,338 @@
+        tasks_response = []
+        for task in tasks: #<Task 2342> - returns object, if we want to see attributes we have to 
+            #access the attributes. So we break down objhects in that list in dictionary so we can see each instance's attributes
+            tasks_response.append({
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": False if task.completed_at == None else True
+            })
+        
+        sort_param = request.args.get("sort")
+        if sort_param == "asc":
+            tasks_response.sort(key=lambda t: t["title"])
+            
+        if sort_param == "desc":
+            tasks_response.sort(key=lambda t: t["title"], reverse=True)
+
+        return jsonify(tasks_response), 200 #describes respond of an endpoint, once we receive this request, what do we respond with
+#take data we want to respond with and put it in a javascript object format aka JSON, turns data into JSON form b/c JSON is an accepted data type to communicate between two entities
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+        if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+            # return make_response("400 Bad Request")
+            return {"details": "Invalid data"}, 400
+        
+        new_task = Task(
+            title= request_body["title"],
+            description= request_body["description"],
+            completed_at= request_body["completed_at"]
+        )
+        db.session.add(new_task)
+        db.session.commit()
+
+        return {
+            "task": {
+                "id": new_task.task_id,
+                "title": new_task.title,
+                "description": new_task.description,
+                "is_complete": False if new_task.completed_at == None else True
+            }
+        }, 201
+
+
+@tasks_bp.route("/<task_id>", methods= ["GET", "PUT", "DELETE"])
+def handle_task(task_id):
+    task = Task.query.get(task_id)
+    if request.method == "GET":
+        if task is None:
+            return jsonify(None), 404
+        elif task.goal_id == None:
+            return {
+                "task": {
+                    "id": task.task_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": False if task.completed_at == None else True
+                }
+            }
+        else:
+            # return jsonify(task), 200
+             return {
+                "task": {
+                    "id": task.task_id,
+                    "goal_id": task.goal_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": False if task.completed_at == None else True
+                }
+            }
```

Comment:
Instead of recreating the entire dictionary again when a `goal_id` is present, we can create a base dictionary then add a goal_id key if present in the base dictionary.

```suggestion
        task_dict =  {
                "task": {
                    "id": task.task_id,
                    "title": task.title,
                    "description": task.description,
                    "is_complete": False if task.completed_at == None else True
                }
            }
        if task.goal_id:
             task_dict["goal_id"] = task.goal_id
             return task_dict, 200
        else:
             task_dict["goal_id"] = task.goal_id
             return task_dict, 200
```